Sources/TuistGenerator/Generator/ConfigGenerator.swift

```diff
@@ -191,5 +191,24 @@ final class ConfigGenerator: ConfigGenerating {
                 }
             }
         }
+
+        if let deploymentTarget = target.deploymentTarget {
```

Comment:
Any help how we could test it? Shall we add checking the settings build to an existing method in `ConfigGeneratorTest` or maybe create a new one?

Reply:
Sure, here's an example of one of the test cases that could be written in `ConfigGeneratorTests` 

```swift
func test_generateTargetWithDeploymentTarget() throws {
        // Given
        let target = Target.test(deploymentTarget: .iOS("12.0", [.iphone, .ipad]))
        
        // When
        try subject.generateTargetConfig(target,
                                     pbxTarget: pbxTarget,
                                     pbxproj: pbxproj,
                                     projectSettings: .default,
                                     fileElements: ProjectFileElements(),
                                     graph: Graph.test(),
                                     sourceRootPath: AbsolutePath("/project"))
        
        // Then
        let configurationList = pbxTarget.buildConfigurationList
        let debugConfig = configurationList?.configuration(name: "Debug")
        let releaseConfig = configurationList?.configuration(name: "Release")
        
        let expectedSettings = [
            "TARGETED_DEVICE_FAMILY": "1,2",
            "IPHONEOS_DEPLOYMENT_TARGET": "12.0"
        ]
        
        assert(config: debugConfig, contains: expectedSettings)
        assert(config: releaseConfig, contains: expectedSettings)
    }
```

Reply:
I added deployment target tests to `ConfigGenerator`, thanks for your help 🙇